lib/Conversion/PolynomialToStandard/PolynomialToStandard.cpp

```diff
@@ -682,18 +690,19 @@ func::FuncOp PolynomialToStandard::buildPolynomialModFunc(FunctionType funcType,
   // The result remainder is still in the larger ring, but needs to have its
   // coefficients modded and then converted to the smaller ring.
   auto toTensorOp = builder.create<ToTensorOp>(inputType, whileOp.getResult(0));
-  auto remainderModOp = builder.create<arith::ConstantOp>(
-      inputType,
-      DenseElementsAttr::get(inputType, ring.getCmod().getZExtValue()));
-  auto remainderCoeffsModded = builder.create<arith::RemUIOp>(
-      toTensorOp.getResult(), remainderModOp.getResult());
+  auto remainderModArg = builder.create<arith::ConstantOp>(
+      inputType, DenseElementsAttr::get(
+                     inputType, APInt(inputType.getElementTypeBitWidth(),
+                                      ring.getCmod().getZExtValue())));
```

Comment:
here again, correct me if i'm wrong but zero extending the cmod seems like it would change its value if you're using it for a signed remainder below

Reply:
Done.